### ISSUE TITLE
Nested query filtering

### ISSUE THREAD

**Issue by jeggy** [graphql/graphql-js#1047]
How can one get the result of more than just one level up and filter by multiple levels?

[I created minimal example repository showing this problem](https://github.com/jeggy/graphql-nested-problem)

For example with this query:
```graphql
{
  country(id: 2) {
    id
    name
    stores {
      id
      name
      cars {
        id
        model
        storeId
        countryId
}}}}
```

I only want to get the cars that are connected to that store and country. But the country query is two levels up, so I don't have access to the result of it in the resolver.

**Comment by lxjwlt** [comment]
+1  facebook/graphql#88

**Comment by leebyron** [comment]
Ensure the values returned from `stores` includes information about the country they relate to, such that when querying `cars` on each `store`, you'll have that context available.

Alternatively, if each `store` is unique and may appear in multiple `country`, then `cars` will only be relative to each unique `store`. If you care to filter them down to country, perhaps you should use a field argument like `cars(inCountry: 2)`

In other words, this should not be solved by looking back in the query but instead by making sure you have the appropriate data in your data model.
